File: app/src/main/java/com/example/assignment/utils/Convert.java
package com.example.assignment.utils;

import android.content.Context;
import android.view.View;

public class Convert {
    public static float pixelsToSp(Context context, float px) {
        float scaledDensity = context.getResources().getDisplayMetrics().scaledDensity;
        return px / scaledDensity;
    }

    public static int dpToPixel(View v, int dp) {
        return (int) (dp * v.getContext().getResources().getDisplayMetrics().density);
    }

    public static int pixelToDp(View v, int pixel) {
        return (int) (pixel / v.getContext().getResources().getDisplayMetrics().density);
    }

}

File: app/src/main/java/com/example/assignment/utils/Convert.kt
package com.example.assignment.utils

import android.content.Context
import android.view.View

object Convert {
    fun pixelsToSp(context: Context, px: Float): Float {
        val scaledDensity = context.resources.displayMetrics.scaledDensity
        return px / scaledDensity
    }

    fun dpToPixel(v: View, dp: Int): Int {
        return (dp * v.context.resources.displayMetrics.density).toInt()
    }

    fun pixelToDp(v: View, pixel: Int): Int {
        return (pixel / v.context.resources.displayMetrics.density).toInt()
    }
}